Kernel/Thread.h

```diff
@@ -264,6 +264,7 @@ class Thread {
     bool is_stopped() const { return m_state == Stopped; }
     bool is_blocked() const { return m_state == Blocked; }
     bool in_kernel() const { return (m_tss.cs & 0x03) == 0; }
+    bool is_dead() const { return m_state == Dead || m_state == Dying; }
```

Comment:
`Dead` and `Dying` are two distinct states, so let's not confuse things by having a getter that makes it look like the states are equivalent.